test/UnitTestHelpers.hs

```diff
@@ -0,0 +1,359 @@
+    , gquit = Nothing
+    , _gleader = Nothing
+    , gstash = Nothing
+    , gvictims = EM.empty
+    }
+
+testActor :: Actor
+testActor = 
+  Actor
+  { btrunk = toEnum 0
+  , bnumber = Nothing
+  , bhp = 0
+  , bhpDelta = ResDelta (0,0) (0,0)
+  , bcalm = 0
+  , bcalmDelta = ResDelta (0,0) (0,0)
+  , bpos = Point 0 0
+  , boldpos = Nothing
+  , blid = testLevelId
+  , bfid = testFactionId
+  , btrajectory = Nothing
+  , borgan = EM.empty
+  , beqp = EM.empty
+  , bweapon = 0
+  , bweapBenign = 0
+  , bwatch = WWatch
+  , bproj = False
+  }
+
+testItemKind :: ItemKind 
+testItemKind = ItemKind
+  { isymbol  = 'x'
+  , iname    = "12345678901234567890123"
+  , ifreq    = [ (UNREPORTED_INVENTORY, 1) ]
+  , iflavour = zipPlain [Green]
+  , icount   = 1 + 1 `Dice.d` 2
+  , irarity  = [(1, 50), (10, 1)]
+  , iverbHit = "hit"
+  , iweight  = 300
+  , idamage  = 1 `Dice.d` 1
+  , iaspects = [ AddSkill Ability.SkHurtMelee $ -16 * 5
+                , SetFlag Ability.Fragile
+                , toVelocity 70 ]
+  , ieffects = []
+  , idesc    = "A really cool test item."
+  , ikit     = []
+  }
+
+testActorWithItem :: Actor
+testActorWithItem = 
+  testActor { beqp = EM.singleton testItemId (1,[])}
+
+-- stublike state instance that should barely function for testing
+stubState :: State
+stubState = let singletonFactionUpdate _ = EM.singleton testFactionId testFaction
+                singletonDungeonUpdate _ = EM.singleton testLevelId stubLevel
+                singletonActorDUpdate _ = EM.singleton testActorId testActor
+                singletonActorMaxSkillsUpdate _ = EM.singleton testActorId Ability.zeroSkills
+                copsUpdate oldCOps = oldCOps{corule=((corule oldCOps){rWidthMax=testLevelDimension, rHeightMax=testLevelDimension})}
+                stateWithMaxLevelDimension = updateCOpsAndCachedData copsUpdate emptyState
+                stateWithFaction = updateFactionD singletonFactionUpdate stateWithMaxLevelDimension
+                stateWithActorD = updateActorD singletonActorDUpdate stateWithFaction
+                stateWithActorMaxSkills = updateActorMaxSkills singletonActorMaxSkillsUpdate stateWithActorD
+                stateWithDungeon = updateDungeon singletonDungeonUpdate stateWithActorMaxSkills
+            in stateWithDungeon
+
+testStateWithItem :: State
+testStateWithItem = let swapToItemActor _ = EM.singleton testActorId testActorWithItem
+                     in updateActorD swapToItemActor stubState
+
+emptyCliState :: CliState
+emptyCliState = CliState
+  { cliState = emptyState 
+  , cliClient = emptyStateClient $ toEnum 0
+  , cliSession = Nothing
+  -- , cliDict = undefined 
+  -- , cliToSave = undefined 
+  }  
+
+stubSessionUI :: SessionUI
+stubSessionUI = (emptySessionUI stubUIOptions) 
+  { sactorUI = EM.singleton testActorId ActorUI { bsymbol='j', bname="Jamie", bpronoun="he/him", bcolor=BrCyan }
+  , sccui = emptyCCUI { coscreen = ScreenContent { rwidth = 10 -- unit test expects rheight > 1
```

Comment:
It wasn't available when you started, but now the `coscreen` inside `emptyCCUI` is similar to what you have here and is validated with `Game.LambdaHack.Client.UI.Content.Screen.validateSingle`, which should prevent UI functions looping, etc. due to too trivial values.

Reply:
Actually, let me see how the tests fail with that change and then I will fix functions called by `splitOKX` to avoid the failures.

Reply:
Strangely the test doesn't fail. I will fix the functions and make validation more strict anyway and when the tests are extended, let's see how the game copes with, say, fitting two columns of text, both indented, on width 5 screen.